src/nodes/interceptor/eth_sign_transaction.go

```diff
@@ -21,12 +23,6 @@ func (i *Interceptor) ethSignTransaction(ctx context.Context, msg *ethereum.Send
 		return nil, jsonrpc.InvalidParamsError(errors.InvalidParameterError(errMessage))
 	}
 
-	if msg.GasPrice == nil {
```

Comment:
Should we check here the following
```
if msg.isLegacy() {
   if msg.GasPrice == nil {
       // error
   }
} else {
   if msg.GasFeeCap == nil {
       // error
   }
}
```

Reply:
these checks are not correct

Reply:
discussed as gasPrice = 0 is valid and maxFee = 0 as well.